src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -226,16 +225,23 @@ private function controlPagedResult($con, int $pageSize, string $cookie): bool
     /**
      * Retrieve LDAP pagination cookie.
      *
-     * @param resource $con
-     * @param resource $result
+     * @param resource   $con
+     * @param resource|0 $result
      */
     private function controlPagedResultResponse($con, $result, string $cookie = ''): string
     {
+        $this->serverctrls = [];
+
         if (\PHP_VERSION_ID < 70300) {
             ldap_control_paged_result_response($con, $result, $cookie);
 
             return $cookie;
         }
+
+        if (0 === $result) {
```

Comment:
to really be compatible and seamless, this should be an `elseif` - otherwise, we're affecting the behavior on php <=7.3.

a test case would really help if you can have a look

Reply:
There's a return in the previous if content, how can this impact the previous statements? 🤔 

I was not able to run the test entire test suite on my computer :( . I'll give another try, but if you have any insights about running slapd that'd be great ^^' .